### DISCUSSION TITLE
Set layer name for PSD files

### DISCUSSION THREAD

**Discussion by cody-SW** [ImageMagick/ImageMagick#5421] [Digital Image Processing] [answered]
Hi, 
I would like to convert a jpeg to psd with a single  layer named Background. This does the conversion but the layer is named L1

In Python 3:
os.system(f"convert UAMFSS-OR-1.jpg UAMFSS-OR-1.psd")

Does anyone know how to change the default layer name L1 to Background?

I have tried this:
os.system(f"convert UAMFSS-OR-1.jpg label:Background UAMFSS-OR-1.psd")

This will name the layer Background, but then it is just an empty transparency.
![Screen Shot 2022-08-05 at 6 34 02 PM](https://user-images.githubusercontent.com/62605701/183121735-3d5d8abd-c58c-484c-bd6f-f2cceda9a5af.png)
![Screen Shot 2022-08-05 at 6 34 16 PM](https://user-images.githubusercontent.com/62605701/183121748-39d3c438-515b-4d85-af3a-dfd06d0d862a.png)

**Comment by fmw42** [comment]
I do not believe that Imagemagick can name PSD layers at this time. One of the IM developers should reply to confirm or correct me.

**Comment by dlemstra** [comment]
`label:Background` should be `-label:Background`.

**Reply by cody-SW** [reply]
I get this error when I run as -label:Background
convert: unrecognized option `-label:Background' @ error/convert.c/ConvertImageCommand/2096.

**Reply by snibgo** [reply]
Try `-label Background`

**Reply by cody-SW** [reply]
no error this way, but the layer is still named L1

**Comment by snibgo** [comment, marked as answer]
Try `-set label Background`

**Reply by cody-SW** [reply]
Thank you! That works!

**Comment by lmiol** [comment]
but how to name if there is a few layers
like this for example

magick BG_Lagoon.jpg -set label Backgroundnew Platform_Lagoon1A.png -set label Background ( -clone 0--1 -background none -flatten ) +insert BG_HBCross_new.psd

it will rename both to Background, but how name each of layers separatly?

UPDATE
find out
magick  ( -label "top" -page 280x228+0+0 Platform_Lagoon1A.png -background none ) ( -label "middle" -page 280x228+969+366 Platform_Lagoon1B.png -background none  ) ( -label "background" -page 2048x1151+0+0 BG_Lagoon.jpg -mosaic ) ( -clone 0--1 -mosaic ) -reverse "output.psd"
